adapters/xeworks/xeworks.go

```diff
@@ -0,0 +1,156 @@
+package xeworks
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := new(adapter)
+	bidder.endpoint = template
+
+	return bidder, nil
+}
+
+func (a *adapter) buildEndpointFromRequest(bidRequest *openrtb2.BidRequest) (string, error) {
+	impExtRaw := bidRequest.Imp[0].Ext
+	var impExt adapters.ExtImpBidder
+
+	if err := json.Unmarshal(impExtRaw, &impExt); err != nil {
+		return "", &errortypes.BadInput{
+			Message: "Bidder impression extension can't be deserialized",
+		}
+	}
+
+	var xeworksExt openrtb_ext.ExtXeworks
+	if err := json.Unmarshal(impExt.Bidder, &xeworksExt); err != nil {
+		return "", &errortypes.BadInput{
+			Message: "Xeworks extenson can't be deserialized",
+		}
+	}
+
+	endpointParams := macros.EndpointTemplateParams{
+		Host:     xeworksExt.Env,
+		SourceId: xeworksExt.Pid,
+	}
+
+	return macros.ResolveMacros(a.endpoint, endpointParams)
+}
+
+func (a *adapter) MakeRequests(
+	openRTBRequest *openrtb2.BidRequest,
+	requestInfo *adapters.ExtraRequestInfo,
+) ([]*adapters.RequestData, []error) {
+	endpoint, err := a.buildEndpointFromRequest(openRTBRequest)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	body, err := json.Marshal(openRTBRequest)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	return []*adapters.RequestData{{
+		Method:  http.MethodPost,
+		Body:    body,
+		Uri:     endpoint,
+		Headers: headers,
+	}}, nil
+}
+
+func (a *adapter) MakeBids(
+	openRTBRequest *openrtb2.BidRequest,
+	requestToBidder *adapters.RequestData,
+	bidderRawResponse *adapters.ResponseData,
+) (*adapters.BidderResponse, []error) {
```

Comment:
nit - requesting to please write function definition in single line (in-line with other functions) to maintain consistency.